piecedirectory/doctor.go

```diff
@@ -135,10 +142,15 @@ func (d *Doctor) checkPiece(ctx context.Context, pieceCid cid.Cid, lu *sectorsta
 		}
 	}
 
+	if !hasDealsOnThisMiner {
+		doclog.Debugw("ignoring piece as it is not present in any deals on this miner", "piece", pieceCid.String(), "miner", d.maddr.String())
+		return nil
+	}
```

Comment:
I'm confused -- we go over all pieces stored in LID on all boostd instances?

Reply:
This is just a sanity check. There are two different processes that populate the list of pieces to check, and that add / remove deals. So it's possible that
- the process that adds to the list of pieces to check adds the piece from deal A
- another process deletes deal A
- GetNextPiecesToCheck returns the piece cid from deal A
- there are no deals for that piece

Reply:
You're right that this should happen very rarely. I updated the log level to a warning.